### PR TITLE
Enable bg_assertsAutoBreak by default

### PR DESCRIPTION
**Pull request by cgalvan**
## What does this PR do?

This change will enable `bg_assertsAutoBreak` by default, which will allow developers to catch issues flagged by asserts quicker. Most recently, a change went in that passed AR but had multiple duplicated Uuid's for classes:

```
System: Trace::Assert
 D:\github\o3de\Code\Framework\AzCore\AzCore/Serialization/SerializeContext.h(1977): (49704) 'class AZ::SerializeContext::ClassBuilder __cdecl AZ::SerializeContext::Class<class AWSClientAuth::AWSClientAuthSystemComponent,class AZ::Component>(class AZ::SerializeContext::IObjectFactory *)'
System: This class type AWSClientAuthSystemComponent could not be registered with duplicated Uuid: {0C2660C8-1B4A-4474-BE65-B487E2DE8649}.
```
This could've been caught before the change was merged if `bg_assertsAutoBreak` was on by default.

## How was this PR tested?

Ran latest development (that currently has several asserts triggering) and verified that the Editor now breaks on the assert.

### CODE REVIEW

**Comment by cgalvan** [comment]
NOTE: I won't merge this change until I have fixed/merged the assert case mentioned above.

**Review by gadams3 — APPROVED** [review]

**Review by dtamkin1 — APPROVED** [review]

**Review by mbalfour-amzn — APPROVED** [review]
I have no idea why we haven't made this the default from the start. :)

**Review by moraaar — APPROVED** [review]

**Review by lsemp3d — APPROVED** [review]

**Review by lawsonamzn — APPROVED** [review]

**Review by amzn-tmryan — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**
